### PR TITLE
Issue 5886: SLTS - Fix several flaky tests. Use multiple threads in tests to uncover more thread/timing related issues.

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
Issue 5886: SLTS - Fix several flaky tests. Use multiple threads in tests to uncover more thread/timing related issues.

**Purpose of the change**  
Fixes https://github.com/pravega/pravega/issues/5886 , https://github.com/pravega/pravega/issues/6024

**What the code does**  
Fix possible null exception.
Use multiple threads in tests to uncover more thread/timing related issues.
Fix issues where test were not waiting on futures

**How to verify it**  
Tests must pass

### CODE REVIEW

File: segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageTests.java [hunk 1]
```diff
@@ -68,6 +69,11 @@ public void after() throws Exception {
         super.before();
     }
 
+    @Override
+    protected int getThreadPoolSize() {
+        return THREAD_POOL_SIZE;
```
Comment (andreipaduroiu) [inline segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageTests.java:74]: Why do you need 10 threads for this? 1-3 is more than sufficient.

GH Actions doesn't give you more than 2 processors, and you may barely be able to use one of them since we run our tests in parallel. I don't understand the benefit that running your tests with 10 threads may bring.

Comment (andreipaduroiu) [reply segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageTests.java:74]: Same everywhere else below.

**Review by andreipaduroiu — CHANGES_REQUESTED** [review]

File: segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageTests.java (hunk 1 above)
Comment (sachin-j-joshi) [reply segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageTests.java:74]: My intention here is to add more randomness in task scheduling. With more threads the probability of getting same scheduling every time decreases while increasing the probability of hitting corner case related to timing issues.

Comment (sachin-j-joshi) [reply segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageTests.java:76]: Fixed. Using 3 threads.

File: segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/metadata/ChunkMetadataStoreTests.java [hunk 2]
```diff
@@ -54,13 +54,19 @@
     protected static final String VALUE5 = "avian";
     protected static final String VALUE6 = "bird";
 
+    private static final int THREAD_POOL_SIZE = 10;
```
Comment (sachin-j-joshi) [inline segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/metadata/ChunkMetadataStoreTests.java:57]: 10 threads are required.

**Review by andreipaduroiu — APPROVED** [review]
